File: examples/python/snippets/document/document_cleanup.py
import os
from typing import Optional

from scanbotsdk import *
from utils import create_image_ref


def cleanup_document(image_path: str, mask_path: str, save_path: Optional[str] = None):
    with create_image_ref(image_path) as image, create_image_ref(mask_path) as mask:
        cleanup = DocumentCleanup(
            image=image,
            configuration=DocumentCleanupConfiguration()
        )
        result = cleanup.run(mask=mask)

        print(f"Document cleanup status: {result.status.name}")

        if result.status == DocumentCleanupStatus.OK:
            original_info = image.info()
            cleaned_info = result.image.info()
            print(f"Original WxH: {original_info.width}x{original_info.height}")
            print(f"Cleaned  WxH: {cleaned_info.width}x{cleaned_info.height}")

            if save_path:
                result.image.save_image(save_path, SaveImageOptions())
                print(f"Saved cleaned image to: {os.path.abspath(save_path)}")

File: examples/nodejs/src/snippets/document/document-cleanup.ts
import * as ScanbotSDK from "scanbotsdk";

export class DocumentCleanupSnippet {
  public static async run(filePath: string, maskPath: string, savePath?: string): Promise<void> {
    await using image = await ScanbotSDK.ImageRef.fromPath(filePath);
    await using mask = await ScanbotSDK.ImageRef.fromPath(maskPath);
    await using cleanup = await ScanbotSDK.DocumentCleanup.create(
      image,
      new ScanbotSDK.DocumentCleanupConfiguration()
    );
    await using result = await cleanup.run(mask);

    console.log("Document cleanup status: " + result.status);

    const originalInfo = await image.imageInfo();
    const cleanedInfo = await result.image.imageInfo();

    console.log(`Original WxH: ${originalInfo.width}x${originalInfo.height}`);
    console.log(`Cleaned  WxH: ${cleanedInfo.width}x${cleanedInfo.height}`);

    if (savePath) {
      await result.image.saveImage(savePath, new ScanbotSDK.SaveImageOptions());
      console.log("Saved cleaned image to: " + savePath);
    }
  }
}
